src/Workspaces/CSharp/Portable/Extensions/ExpressionSyntaxExtensions.cs

```diff
@@ -1114,10 +1126,67 @@ private static int GetNamespaceId(SyntaxList<MemberDeclarationSyntax> members, N
             return -1;
         }
 
+        private static bool TryNormalizeOrReduce(
+            this NameSyntax name,
+            SemanticModel semanticModel,
+            out ExpressionSyntax replacementNode,
+            out TextSpan issueSpan,
+            OptionSet optionSet,
+            CancellationToken cancellationToken)
+        {
+            if (name.TryNormalize(semanticModel, out replacementNode, out issueSpan, optionSet))
+            {
+                return true;
+            }
+
+            return name.TryReduce(semanticModel, out replacementNode, out issueSpan, optionSet, cancellationToken);
+        }
+
+        private static bool TryNormalize(
+            this NameSyntax name,
+            SemanticModel semanticModel,
+            out ExpressionSyntax replacementNode,
+            out TextSpan issueSpan,
+            OptionSet optionSet)
+        {
+            replacementNode = null;
+            issueSpan = default(TextSpan);
+
+            // if we can't resolve the symbol then nothing can be inferred
+            var symbol = SimplificationHelpers.GetOriginalSymbolInfo(semanticModel, name);
+            if (symbol == null)
+            {
+                return false;
+            }
+
+            // if not static and not already part of a `this.` expression
+            if (!symbol.IsStatic &&
+                name is SimpleNameSyntax &&
+                (name.Parent as MemberAccessExpressionSyntax)?.Expression.Kind() != SyntaxKind.ThisExpression)
+            {
+                if ((symbol.IsKind(SymbolKind.Field) && optionSet.GetOption(SimplificationOptions.QualifyMemberFieldAccessWithThisOrMe, semanticModel.Language)) ||
+                    (symbol.IsKind(SymbolKind.Property) && optionSet.GetOption(SimplificationOptions.QualifyMemberPropertyAccessWithThisOrMe, semanticModel.Language)) ||
+                    (symbol.IsKind(SymbolKind.Method) && optionSet.GetOption(SimplificationOptions.QualifyMemberMethodAccessWithThisOrMe, semanticModel.Language)) ||
+                    (symbol.IsKind(SymbolKind.Event) && optionSet.GetOption(SimplificationOptions.QualifyMemberEventAccessWithThisOrMe, semanticModel.Language)))
+                {
+                    replacementNode =
+                        SyntaxFactory.MemberAccessExpression(
+                            SyntaxKind.SimpleMemberAccessExpression,
+                            SyntaxFactory.ThisExpression(),
+                            (SimpleNameSyntax)name.WithLeadingTrivia().WithoutAnnotations(Simplifier.Annotation))
+                        .WithLeadingTrivia(name.GetLeadingTrivia());
+                    issueSpan = name.Span;
+                    return true;
```

Comment:
I don't think this is the right thing to do here. The simplifier shouldn't introducing new nodes. It's job is to remove them. The way this should work is that the Expander should add `this.` in all cases (it already does) and the simplifier should remove them based on the current options.